lib/livebook_web/live/session_live/app_settings_component.ex

```diff
@@ -0,0 +1,156 @@
+        <div class="flex flex-col space-y-4">
+          <.text_field field={f[:slug]} label="Slug" spellcheck="false" phx-debounce />
+          <div class="flex flex-col space-y-1">
+            <.checkbox_field
+              field={f[:access_type]}
+              label="Password-protected"
+              checked_value="protected"
+              unchecked_value="public"
+            />
+            <%= if Ecto.Changeset.get_field(@changeset, :access_type) == :protected do %>
+              <.password_field field={f[:password]} spellcheck="false" phx-debounce />
+            <% end %>
+          </div>
+          <.radio_button_group_field
+            field={f[:multi_session]}
+            options={[{"false", "Single"}, {"true", "Multi"}]}
+            label="Session type"
+            help={
+              ~S'''
+              Whether all users should share the
+              same session or be able to manually
+              start multiple sessions.
+              '''
+            }
+            full_width
+          />
+          <.select_field
+            field={f[:auto_shutdown_ms]}
+            label="Shutdown after inactivity"
+            options={[
+              {"Never", ""},
+              {"5 seconds", "5000"},
+              {"1 minute", "60000"},
+              {"1 hour", "3600000"}
+            ]}
+            help={
+              ~S'''
+              Shuts down the session if it has no
+              users for the specified amount of time
+              '''
+            }
+          />
+          <.checkbox_field
+            field={f[:show_source]}
+            label="Show source"
+            help={
+              ~S'''
+              When enabled, makes notebook source
+              accessible in the app menu
+              '''
+            }
+          />
+          <.checkbox_field
+            field={f[:output_type]}
+            label="Only render rich outputs"
+            checked_value="rich"
+            unchecked_value="all"
+            help={
+              ~S'''
+              When enabled, hides simple outputs
+              and only shows rich elements, such
+              as inputs, frames, tables, etc.
+              '''
+            }
+          />
+          <%= if Ecto.Changeset.get_field(@changeset, :multi_session) do %>
+            <.checkbox_field
+              field={f[:show_existing_sessions]}
+              label="List existing sessions"
+              help={
+                ~S'''
+                When enabled, shows all existing
+                sessions on the app page
+                '''
+              }
+            />
+          <% else %>
+            <.checkbox_field
+              field={f[:zero_downtime]}
+              label="Zero-downtime deployment"
+              help={
+                ~S'''
+                When enabled, app keeps pointing
+                to an old session, until the newly
+                deployed notebook is fully executed
```

Comment:
```suggestion
                When enabled, the application is only
                deployed after it executes all of its
                cells, making sure the application is
                ready for use.
```

Reply:
The app is already deployed, it's the new version (?)

```suggestion
                When enabled, the new version is
                deployed after it executes all of
                its cells, making sure it is ready
                for use.
```

Reply:
In the current released version, even if there is no previous version, i can only access the new version once it finishes booting. So to me it applies regardless of old/new versions?

Reply:
Ah, and we are forgetting to say it doesn't deploy at all if it fails to boot. So maybe this:

```suggestion
                When enabled, the app becomes accessible
                only after it executes all of its cells,
                making sure it is ready for use.
                If an error happens, deploy is aborted.
```

Reply:
Actually, currently it would direct to the session even if booting/errored if it is the first deployment. The zero-downtime is more about publishing a new version. But we can change that.

Reply:
Alright, after our conversation:

```suggestion
                When enabled, a new version only becomes
                available after it executes all of its
                cells, making sure it is ready to use.
                If an error happens, deploy is aborted.
```